pylabrobot/tilting/tilter.py

```diff
@@ -1,89 +1,213 @@
+      self.assign_resource_to_site(resource, position=idx)
+
+  def __delitem__(self, idx):
+    """Unassign a resource from this tilter. See :meth:`~Tilter.unassign_child_resource`"""
+    self.unassign_child_resource(self[idx])
 
-    We assume the rotation anchor is the right side of the module. This may change in the future
-    if we integrate other tilt modules.
+  def get_resources(self) -> List[Resource]:
+    """Get all resources, using self.__getitem__ (so that the location is within this tilter)."""
+    return [self[idx] for idx in range(self.capacity) if self[idx] is not None]
+
+  @property
+  def absolute_angle(self) -> int:
+    return self._absolute_angle
+
+  async def set_angle(self, absolute_angle: int):
+    """Set the tilt module to rotate to a given angle.
+    Args:
+      angle: The absolute angle to set rotation to, in degrees, measured from horizontal as zero.
+    """
+    # if the hinge is on the left side of the tilter, the angle is kept positive
+    # else, the angle is converted to negative. this follows Euler angle conventions.
+    # This conversion is done to simplify coordinate calculations.
+    angle = absolute_angle if self._hinge_coordinate.x < self._size_x / 2 else -absolute_angle
+    await self.backend.set_angle(angle=abs(angle))
+    self._absolute_angle = absolute_angle
+
+  def rotate_coordinate_around_hinge(self, absolute_coordinate: Coordinate, angle: int) -> Coordinate:
+    """
+    Rotate an absolute coordinate around the hinge of the tilter by a given angle.
 
     Args:
-      angle: The angle to rotate by, in degrees. Clockwise. 0 is horizontal.
+      absolute_coordinate (Coordinate): The coordinate to rotate.
+      angle (int): The angle to rotate by, in degrees. Negative is clockwise according to Euler conventions.
+
+    Returns:
+      Coordinate: The new coordinate after rotation.
     """
+    theta = math.radians(angle)
 
-    await self.backend.set_angle(angle=angle)
+    rotation_arm_x = absolute_coordinate.x - (
+      self._hinge_coordinate.x + self.get_absolute_location("l", "f", "b").x
+    )
+    rotation_arm_z = absolute_coordinate.z - (
+      self._hinge_coordinate.z + self.get_absolute_location("l", "f", "b").z
+    )
 
-    for well in self.get_plate().children:
-      assert well.location is not None
+    x_prime = rotation_arm_x * math.cos(theta) - rotation_arm_z * math.sin(theta)
+    z_prime = rotation_arm_x * math.sin(theta) + rotation_arm_z * math.cos(theta)
 
-      # Convert angle to radians.
-      theta = math.radians(angle)
+    new_x = x_prime + (self._hinge_coordinate.x + self.get_absolute_location("l", "f", "b").x)
+    new_z = z_prime + (self._hinge_coordinate.z + self.get_absolute_location("l", "f", "b").z)
 
-      # Compute the current location of the well. Use the rotation anchor as the origin.
-      x = self.get_plate().get_size_x() - well.location.x
-      z = well.location.z
-      h = math.sqrt(x**2 + z**2) # hypotenuse (dist from anchor to well)
```

Comment:
Git diff started messing up here, but basically I removed the part where it updates the well origins and replaced it with a function that returns asp/disp offsets. If you want, we can revert the changes and keep updating the well locations, but I don't think it's super helpful right now since we can't virtually "angle" the wells so that `.get_absolute_location` makes sense (as far as I'm aware, maybe this functionality does exist :D).